app/controllers/users_controller.rb

```diff
@@ -10,7 +10,8 @@ def index
   # GET /users/1
   # GET /users/1.json
   def show
-    @user_events_created = @user.events_created.all
+    # @user_events_created = @user.events_created.all
+    @user_events = User.find(params[:id])
```

Comment:
The name of the instance variable is misleading because this is the `user` record pertaining to a given `id`. Please use `user.events` or `user.created_events` to refer to all events created by a given user. Don't rely on calling `@user_events.events_created.each` on your view; instead have this query already stored in your controller and then refer to it in your view [`REQUIRED`].

Reply:
Furthermore, use the event model scope, i.e. `@past_events = current_user.attended_events.previous` to show upcoming and previous events / attendance for a user. Favor having this in your controller as opposed to your view [`REQUIRED`].